### ISSUE TITLE
Task 16: Calculate gain/loss (amount and percentage)

### ISSUE THREAD

**Issue by vongohren** [vongohren/FinanceControll#18]
# Calculate Gain/Loss

**Parent Epic:** #1
**Phase:** 2 - Calculations
**Depends on:** #16, #17 (Cost basis and current value)

## Formula

```
gain_loss = current_value - cost_basis
gain_loss_percent = (gain_loss / cost_basis) * 100
```

## Implementation

```typescript
// lib/calculations/gain-loss.ts

interface GainLossResult {
  costBasis: number;
  currentValue: number;
  gainLoss: number;
  gainLossPercent: number;
  isGain: boolean;
  hasValuation: boolean;
}

export function calculateGainLoss(
  costBasis: number,
  currentValue: number | null
): GainLossResult {
  if (currentValue === null) {
    return {
      costBasis,
      currentValue: 0,
      gainLoss: 0,
      gainLossPercent: 0,
      isGain: false,
      hasValuation: false,
    };
  }

  const gainLoss = currentValue - costBasis;
  const gainLossPercent = costBasis > 0 
    ? (gainLoss / costBasis) * 100 
    : 0;

  return {
    costBasis,
    currentValue,
    gainLoss,
    gainLossPercent,
    isGain: gainLoss >= 0,
    hasValuation: true,
  };
}

// Combined hook
export function useAssetMetrics(assetId: string) {
  const costBasis = useCostBasis(assetId);
  const currentValue = useCurrentValue(assetId, costBasis?.totalQuantity ?? 0);
  
  return useMemo(() => ({
    ...costBasis,
    ...currentValue,
    ...calculateGainLoss(
      costBasis?.totalCostBasis ?? 0,
      currentValue?.totalValue ?? null
    ),
  }), [costBasis, currentValue]);
}
```

## Display Component

```typescript
// components/gain-loss-badge.tsx

export function GainLossBadge({ gainLossPercent, isGain }: { 
  gainLossPercent: number; 
  isGain: boolean;
}) {
  return (
    <Badge className={isGain ? 'bg-green-100 text-green-800' : 'bg-red-100 text-red-800'}>
      {isGain ? '↑' : '↓'} {Math.abs(gainLossPercent).toFixed(2)}%
    </Badge>
  );
}
```

## AI Instructions

1. Create `lib/calculations/gain-loss.ts`
2. Create combined `useAssetMetrics` hook
3. Create `GainLossBadge` component
4. Update asset cards to show gain/loss
5. Color code: green for gains, red for losses
6. Handle edge cases (no transactions, no valuation)
